contracts/extensions/TokenSupplier.sol

```diff
@@ -0,0 +1,191 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./../common/ERC20Extended.sol";
+import "./ColonyExtension.sol";
+
+
+contract TokenSupplier is ColonyExtension {
+
+  uint256 constant ISSUANCE_PERIOD = 1 days;
+  bytes32 constant VOTING_REPUTATION = keccak256("VotingReputation");
+  bytes32 constant VOTING_HYBRID = keccak256("VotingHybrid");
+
+  // Events
+
+  event TokenSupplyCeilingSet(uint256 tokenSupplyCeiling);
+  event TokenIssuanceRateSet(uint256 tokenIssuanceRate);
+  event TokensIssued(uint256 numTokens);
+
+  // Storage
+
+  uint256 tokenSupplyCeiling;
+  uint256 tokenIssuanceRate;
+  uint256 lastPinged;
+  uint256 lastRateUpdate;
+
+  // Modifiers
+
+  modifier initialised() {
+    require(lastPinged > 0, "token-supplier-not-initialised");
+    _;
+  }
+
+  modifier onlyRoot() {
+    require(colony.hasUserRole(msg.sender, 1, ColonyDataTypes.ColonyRole.Root), "token-supplier-caller-not-root");
+    _;
+  }
+
+  modifier onlyContract() {
+    uint256 size;
+    address msgSender = msg.sender;
+    assembly { size := extcodesize(msgSender) }
+    require(size > 0, "token-supplier-caller-must-be-contract");
+    _;
+  }
+
+  // Public
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("TokenSupplier");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+  }
+
+  /// @notice Called when upgrading the extension (currently a no-op)
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension (currently a no-op)
+  function deprecate(bool _deprecated) public override auth {}
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
```

Comment:
Should this (try to) `issueTokens()`? If so, I think it should be a try/catch here so that it can be uninstalled in case of a mistake though.

Reply:
Eh I don't think that's necessary.